### PR TITLE
move off sapm and use otlp by default

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
This PR removes sapm from the default configuration, and use otlp instead.
The PR also introduces a converter that will take the default sapm exporter configuration and replace it with an otlp exporter.

### CODE REVIEW

**Review by pjanotti — COMMENTED** [review]
@atoulme the changes to the MSI added another location with `/v2/trace` when building the default value for `SPLUNK_TRACE_URL`. Could you, please, grep again for `/v2/trace` to cover those?

File: .github/workflows/scripts/win-test-services.ps1 [hunk 1]
```diff
@@ -44,7 +44,9 @@ $expected_svc_env_vars = @{
   "SPLUNK_REALM"            = "$realm";
   "SPLUNK_API_URL"          = "$api_url";
   "SPLUNK_INGEST_URL"       = "$ingest_url";
-  "SPLUNK_TRACE_URL"        = "${ingest_url}/v2/trace";
+  # We no longer test for this as we have moved to OTLP and it breaks upgrade tests.
+  # We have a configuration converter in place to catch old values as part of the collector startup.
+  #"SPLUNK_TRACE_URL"        = "${ingest_url}:443;
```
Comment (pjanotti) [inline .github/workflows/scripts/win-test-services.ps1:49]: Hum, this can be changed in the test, as long as we track the first version doing the switch. Anyway, much better as a follow-up then in this PR.

File: cmd/otelcol/config/collector/gateway_config.yaml [hunk 2]
```diff
@@ -89,9 +89,10 @@ processors:
 
 exporters:
   # Traces
-  sapm:
-    access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
+  otlp:
+    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com:443"
```
Comment (pjanotti) [inline cmd/otelcol/config/collector/gateway_config.yaml:93]: q: why in some places we explicitly have the default HTTPS port and others not? It doesn't matter since 443 the default anyway, just curious if this is intentional or not.

**Review by pjanotti — APPROVED** [review]
Most places have the port `:443` appended to the default value, it will be good if all of them had it - I mean that the default was consistent everywhere, in this case doesn't matter much if the port is included or not.

Other than that, LGTM.

File: .github/workflows/scripts/win-test-services.ps1 (hunk 1 above)
Comment (atoulme) [reply .github/workflows/scripts/win-test-services.ps1:49]: Note 443 is not optional right now.

File: cmd/otelcol/config/collector/gateway_config.yaml (hunk 2 above)
Comment (atoulme) [reply cmd/otelcol/config/collector/gateway_config.yaml:93]: See https://github.com/open-telemetry/opentelemetry-collector/issues/9505 for context: right now the code fails if no port is explicitly provided, which really is a bad outcome.

HTTPS means nothing for grpc endpoints, but is used to indicate that a TLS connection is necessary. See https://github.com/open-telemetry/opentelemetry-specification/pull/1729 for more information.

File: cmd/otelcol/config/collector/upstream_agent_config.yaml [hunk 3]
```diff
@@ -164,30 +165,30 @@ service:
       - batch
       - resourcedetection
       #- resource/add_environment
-      exporters: [sapm, signalfx]
+      exporters: [otlp, signalfx]
       # Use instead when sending to gateway
-      #exporters: [otlp, signalfx]
+      #exporters: [otlp/gateway, signalfx]
```
Comment (dmitryax) [inline cmd/otelcol/config/collector/upstream_agent_config.yaml:170]: Why do we stick with these comments in every pipeline? Won't it be clearer to remove `otlp/gateway` and put this comment just in one place: `exporters::otlp`?

File: tests/zeroconfig/windows/testdata/resource_traces/aspnetcore.yaml [hunk 4]
```diff
@@ -2,7 +2,6 @@ resource_spans:
 - attributes:
     deployment.environment: zc-iis-test
     host.name: <ANY>
-    os.type: windows
```
Comment (dmitryax) [inline tests/zeroconfig/windows/testdata/resource_traces/aspnetcore.yaml:5]: Why is this changed? Is this what was set by the collector, and now we send data directly?

File: cmd/otelcol/config/collector/upstream_agent_config.yaml (hunk 3 above)
Comment (dmitryax) [reply cmd/otelcol/config/collector/upstream_agent_config.yaml:170]: Having two otlp exporters active but using only one of them doesn't seem right

Comment (dmitryax) [reply cmd/otelcol/config/collector/upstream_agent_config.yaml:170]: Submitted https://github.com/signalfx/splunk-otel-collector/pull/4587

File: cmd/otelcol/config/collector/ecs_ec2_config.yaml [hunk 5]
```diff
@@ -122,9 +122,10 @@ processors:
 
 exporters:
   # Traces
-  sapm:
-    access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
+  otlp:
+    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com"
```
Comment (pjanotti) [inline cmd/otelcol/config/collector/ecs_ec2_config.yaml:126]: @atoulme so places like this are broken

File: cmd/otelcol/config/collector/fargate_config.yaml [hunk 6]
```diff
@@ -96,9 +96,10 @@ processors:
 
 exporters:
   # Traces
-  sapm:
-    access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
+  otlp:
+    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com"
```
Comment (pjanotti) [inline cmd/otelcol/config/collector/fargate_config.yaml:100]: @atoulme same here

File: tests/zeroconfig/windows/testdata/resource_traces/aspnetcore.yaml (hunk 4 above)
Comment (pjanotti) [reply tests/zeroconfig/windows/testdata/resource_traces/aspnetcore.yaml:5]: The collector receiving the signals from the instrumentation is not using the default agent configuration anymore, the stripped down config used now doesn't have the `resourcedetection` processor that adds that attribute.

File: cmd/otelcol/config/collector/upstream_agent_config.yaml (hunk 3 above)
Comment (atoulme) [reply cmd/otelcol/config/collector/upstream_agent_config.yaml:170]: thanks
